nbs/01_lhs_parsing.ipynb

```diff
@@ -34,20 +34,21 @@
    "cell_type": "markdown",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #50.        for graph_obj, attr_constraints in constraints.items():
</pre><p>how come there are graph_objects in the key of constraints?</p><p>Something about the data representation is not clear from the documentation.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/DeanLight/graph_rewrite/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
<p>This function actually adds the attributes' type and value to the graph. The name "constraints" is for legacy since it also included the condition thing that was in the transformer, but it is still suitable, as it helps with making sure during the transformer that the same attribute doesn't appear twice with different values. Anyway, we removed the function and moved this addition of type and value into the transformer in a new commit</p><p> </p>

Reply:
<p>it is ok</p>